types/queries.d.ts

```diff
@@ -2,6 +2,8 @@ import {ByRoleMatcher, Matcher, MatcherOptions} from './matches'
 import {SelectorMatcherOptions} from './query-helpers'
 import {waitForOptions} from './wait-for'
 
+type NonEmptyArray<T> = [T, ...T[]]
```

Comment:
What's the lowest TypeScript version that supports this syntax?

Reply:
good point, I don't know, give me a sec

Reply:
The syntax looks supported in 3.0.1, fails in 2.9.2. (what's the oldest version we support?)

The [noUncheckedIndexAccess](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess) check was introduced in 4.1, so it won't make a difference in anything older than that, or anyone with the check disabled.

Reply:
> The syntax looks supported in 3.0.1, fails in 2.9.2. (what's the oldest version we support?)

I usually follow DefinitelyTyped here (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#support-window) i.e. 3.6 is the lowest supported.